### PR TITLE
New storage proposal

### PR DESCRIPTION
**Pull request by shundhammer**
(like just discussed on Jangouts with jsrain, ancorgs, imo, snwint)

### CODE REVIEW

File: src/lib/storage/free_disk_space.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require_relative "./disk_size"
+require "pp"
```
Comment (imobach) [inline src/lib/storage/free_disk_space.rb]: Is it needed?

File: src/lib/storage/free_disk_space.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require_relative "./disk_size"
```
Comment (imobach) [inline src/lib/storage/free_disk_space.rb]: You can write it as `require_relative "disk_size"`.

File: src/lib/storage/partition_creator.rb [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+require "yast"
+require "fileutils"
+require_relative "./storage_manager"
+require_relative "./proposal_settings"
+require_relative "./proposal_volume"
+require_relative "./disk_size"
+require_relative "./space_maker"
+require_relative "./free_disk_space"
+require "pp"
```
Comment (imobach) [inline src/lib/storage/partition_creator.rb]: Same as above.

File: src/lib/storage/free_disk_space.rb (hunk 1 above)
Comment (shundhammer) [inline src/lib/storage/free_disk_space.rb]: Right, here it's not needed. It probably was for a while, but now not any more.

File: src/lib/storage/free_disk_space.rb (hunk 2 above)
Comment (shundhammer) [inline src/lib/storage/free_disk_space.rb]: Fixed.

File: src/lib/storage/partition_creator.rb (hunk 3 above)
Comment (shundhammer) [inline src/lib/storage/partition_creator.rb]: Fixed.

**Pull request merged**